### PR TITLE
feat(stepper, stepper-item): add numberingSystem property

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #4893

## Summary
Adds `numberingSystem` property to `calcite-stepper`. A couple lines in `calcite-stepper-item` will need to be changed due to a refactor in #5466. The small change can go in which ever PR is merged last.

### CODE REVIEW

File: src/components/stepper-item/stepper-item.tsx [hunk 1]
```diff
@@ -193,7 +203,17 @@ export class StepperItem implements InteractiveComponent {
     updateHostInteraction(this, true);
   }
 
+  disconnectedCallback(): void {
+    disconnectLocalized(this);
+  }
+
   render(): VNode {
+    numberStringFormatter.setOptions({
```
Comment (jcfranco) [inline src/components/stepper-item/stepper-item.tsx:211]: Is there another hook we can use to update the options? You could watch for effectiveLocale, but that wouldn't handle changes to the parent's numbering system. Would that be a common scenario, you think?

**Review by jcfranco — APPROVED** [review]
Aside from a considering another way to update the formatter options outside of `render`, this LGTM! 

🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺
🎺🏇🎺🎺🎺🏇🎺🏇🎺🎺🎺🏇🎺🏇🏇🏇🎺🎺🏇🏇🏇🏇🎺🏇🏇🏇🏇🎺🏇🎺
🎺🏇🏇🎺🏇🏇🎺🏇🎺🎺🎺🏇🎺🏇🎺🎺🏇🎺🏇🎺🎺🎺🎺🏇🎺🎺🎺🎺🏇🎺
🎺🏇🎺🏇🎺🏇🎺🏇🎺🎺🎺🏇🎺🏇🏇🏇🎺🎺🏇🎺🏇🏇🎺🏇🏇🏇🎺🎺🏇🎺
🎺🏇🎺🎺🎺🏇🎺🏇🎺🎺🎺🏇🎺🏇🎺🏇🎺🎺🏇🎺🎺🏇🎺🏇🎺🎺🎺🎺🎺🎺
🎺🏇🎺🎺🎺🏇🎺🎺🏇🏇🏇🎺🎺🏇🎺🎺🏇🎺🏇🏇🏇🏇🎺🏇🏇🏇🏇🎺🏇🎺
🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺🎺

File: src/components/stepper-item/stepper-item.tsx (hunk 1 above)
Comment (benelan) [reply src/components/stepper-item/stepper-item.tsx:211]: Yeah I'm not really sure there is a use case for changing locale and numberingSystem besides testing/debugging. I've just been updating the options to be safe lol. 

I guess we could create an internal event or a mutation observer for the numberingSystem prop on the parent and a watcher for effectiveLocale, if you think that's better

Comment (jcfranco) [reply src/components/stepper-item/stepper-item.tsx]: I'm leaning towards only updating the formatter on `effectiveLocale` changes for now.

**Pull request merged**